app/controllers/settings/general.js

```diff
@@ -200,6 +200,13 @@ export default Controller.extend({
                 }
 
                 this.set('settings.facebook', '');
+                if (skipRenderLoop) {
```

Comment:
I was not sure of side-effects of skipping render loop all-together here, but for when we update it directly from CMD+S save action, allowing render loop causes updated values to not save in settings so I added a skip option when its called from `save` 😬

Reply:
I think the `afterRender` was to avoid "x has been modified twice in a single render loop" error caused by our round-a-bout way of setting `""` then setting the url to force the value change to be picked up when the underlying value hasn't changed but we want to show the full url (eg: original input value of "https://facebook.com/ghost", change the input value to "ghost" and click out, the value stays as "ghost" instead of being changed back to the full url because the underlying settings value never changed to override the scratch value.

We can avoid all of that by using `this.settings.notifyPropertyChange('facebook')` which simplifies everything 🙂 I'll push a commit here in a sec because I already have it from testing.

Longer-term, all of this usage of scratch values needs to disappear, I'm sure there is a better way to do this now.

Reply:
>  all of this usage of scratch values needs to disappear, I'm sure there is a better way to do this now.

💯 agreed 😄